src/components/addBooks.js

```diff
@@ -1,16 +1,34 @@
-import React from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
 
-const AddBooks = () => (
-  <form>
-    <input placeholder="ADD NEW BOOK" />
-    <select name="books" id="book">
-      <option value="Science fiction">The God&apos;s Equation</option>
-      <option value="Action">Volvo</option>
-      <option value="Economy">Dollars</option>
-      <option value="Computer Science">DS n ALgo</option>
-    </select>
-    <button type="submit">Add Book</button>
-  </form>
-);
+const AddBooks = () => {
+  const dispatch = useDispatch();
+
+  const submitBookToStore = (e) => {
+    const newBook = {
+      id: uuidv4(),
+      title: e.target.previousElementSibling.previousElementSibling.value,
+      author: '',
+    };
+
+    dispatch(addBook(newBook));
+
+    // dispatch an action and pass it the newBook object (your action's payload)
+  // dispatch(addBook(newBook));
+  };
+  return (
+    <form>
+      <input placeholder="ADD NEW BOOK" />
+      <select name="books" id="book">
+        <option value="Science fiction">The God&apos;s Equation</option>
+        <option value="Action">Volvo</option>
+        <option value="Economy">Dollars</option>
+        <option value="Computer Science">DS n ALgo</option>
+      </select>
+      <input type="button" value="Add Book" onClick={submitBookToStore} />
+    </form>
```

Comment:
- [ ] It will be great to have your form as a controlled form. What I mean by this is you should have an `onChange` function on your input fields that updates the local state every time a user makes any change to the form. This is a major requirement for this project to be approved. To learn more about react forms check [here](https://reactjs.org/docs/forms.html).

Reply:
![image](https://user-images.githubusercontent.com/50904584/146159053-b88e1e7a-77e2-4f70-96f3-ff639eec5767.png)

At the bottom <button onClick={submitBookToStore}>Add Book</button> is mention, so I used that, is this not the right way? Kindly provide more details.

Reply:
done